prac8 solutions/Student/Student.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+#include "../Name/Name.h"
+
+const unsigned MAX_GRADE_COUNT = 7;
+
+class Student {
+private:
+    Name _name;
+//    unsigned _grades[MAX_GRADE_COUNT];
+    unsigned _gradesCount;
+    unsigned _fac;
+
+//    void copyGrades(const unsigned *grades, unsigned gradesCount);
+    void copyFrom(const Student &other);
+    void free();
+
+public:
+    Student() = default;
+    Student(const Student &other);
+    Student &operator=(const Student &other);
+    ~Student();
```

Comment:
you don't need to implement the rule of 4 since you don't have dynamic memory in this class (class Name handles its own dynamic memory)